x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -531,6 +545,11 @@ private Expression pipeIntArgument(Source source, String pipeName, List<BooleanE
             );
         }
 
+        if (expression.dataType() != INTEGER && expression.dataType() != SHORT && expression.dataType() != BYTE) {
```

Comment:
Do we need to keep the SHORT/BYTE type around? We have value that is a non-negative int here.
```suggestion
        if (expression.dataType() != INTEGER) {
```

Reply:
This is, in practice, true (I don't see a way in which a foldable literal other than INTEGER can reach this line), but the contract of the code here is about `expression.dataType().isInteger()` which covers all integer-like data types. I'll keep it as is, maybe at some point we'll add some casting functions (very not likely, but still).